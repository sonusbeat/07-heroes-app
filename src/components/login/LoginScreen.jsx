import { useContext } from "react";
import AuthContext from "../../auth/AuthContext";
import types from '../../types/types';

const LoginScreen = ({ history }) => {

  const { dispatch } = useContext( AuthContext );

  const handleLogin = () => {
    // history.push("/"); // <== Redirecciona a la ruta deseada
    // history.replace("/"); // <== Borra el history

    const lastPath = localStorage.getItem("lastPath") || "/";

    const action = {
      type: types.login,
      payload: {
        name: "Daniel"
      }
    };

    dispatch(action);

    history.replace( lastPath );
  };

  return (
    <section className="container mt-5">
      <h2>Login</h2>
      <hr />

      <button
        className="btn btn-primary"
        onClick={ handleLogin }
      >Acceder</button>
    </section>
  );
}

export default LoginScreen;