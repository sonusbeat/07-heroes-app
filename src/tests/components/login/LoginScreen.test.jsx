import { mount } from 'enzyme';
import AuthContext from '../../../auth/AuthContext';

import LoginScreen from '../../../components/login/LoginScreen';
import types from '../../../types/types';

describe('Pruebas en el componente <LoginScreen />', () => {

  const history = {
    replace: jest.fn(),
  };

  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: false
    }
  };

  const wrapper = mount(
    <AuthContext.Provider value={ contextValue }>
      <LoginScreen history={ history } />
    </AuthContext.Provider>
  );

  test('Deberia de mostrarse correctamente', () => {

    expect( wrapper.html() ).toMatchSnapshot();

  });

  test('Deberia de disparar el dispatch y la navegación', () => {

    // Guardar la referencia handleClick
    const handleClick = wrapper.find("button").prop("onClick");

    // Simular el boton con onClick
    handleClick();

    // Simular el contextValue con dispatch
    expect( contextValue.dispatch ).toHaveBeenCalledWith({
      type: types.login,
      payload: {
        name: "Daniel"
      }
    });

    // Llamar el history.replace
    expect( history.replace ).toHaveBeenCalled();

    // Crear un localStorage
    localStorage.setItem('lastPath', '/dc');

    // Llamar la función handleClick otra vez para guardar el
    // localStorage
    handleClick();

    // Revizar si el modo history se llamo el método replace
    // con el argumento especificado
    expect( history.replace ).toHaveBeenCalledWith("/dc");

  });

});
