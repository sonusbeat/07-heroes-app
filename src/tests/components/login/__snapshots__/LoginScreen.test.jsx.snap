// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <LoginScreen /> Deberia de mostrarse correctamente 1`] = `"<section class=\\"container mt-5\\"><h2>Login</h2><hr><button class=\\"btn btn-primary\\">Acceder</button></section>"`;
